### PR TITLE
Feature/gcp nginx ingress

### PR DESCRIPTION
**Pull request by rusenask**
Changed:
- using nginx ingress for model deployments on GCP
- using precompiled ds-startup file instead of startup.sh

### CODE REVIEW

File: gcp/dotscience-gcp.tf [hunk 1]
```diff
@@ -83,14 +88,22 @@ resource "google_compute_instance" "dotscience_hub_vm" {
...
 #!/bin/bash -xe
 echo "Starting Dotscience hub"
-/home/ubuntu/startup.sh --admin-password "${var.admin_password}" --cloud gcp --hub-size "${var.hub_volume_size}" --hub-device /dev/sdb --hub-hostname "${local.hub_hostname}" --use-kms=false --license-key="${var.license_key}" --letsencrypt-mode="${var.letsencrypt_mode}" --gcp-runner-project "${var.project}" --gcp-runner-zone "${local.zone}" --gcp-runner-machine-type "${var.runner_machine_type}" --deployer-token "${random_id.deployer_token.hex}" --grafana-user "${var.grafana_admin_user}" --grafana-host "${local.grafana_host}"  --grafana-password "${var.grafana_admin_password}"
+
+export GO111MODULE=off
+export GOPATH=/home/ubuntu/go
+go get github.com/spf13/cobra
+
+sudo wget -O /usr/local/bin/ds-startup https://storage.googleapis.com/dotscience-startup/unstable/ds-startup
+sudo chmod +wx /usr/local/bin/ds-startup
```
Comment (prisamuel) [inline gcp/dotscience-gcp.tf:105]: How does this get uploaded? and is it versioned?

Comment (rusenask) [reply gcp/dotscience-gcp.tf:105]: currently here https://github.com/dotmesh-io/dotscience-images/pull/71/files but once it's merged, I will add another step to generate tagged binaries

Comment (rusenask) [reply gcp/dotscience-gcp.tf:105]: so the plan is to have three binaries uploaded:
1. unstable on each commit to master
2. 1.5.0 - tagged ones (so you can pin to a specific version if you want to)
3. stable - latest tagged one (generated on each tag)

Comment (prisamuel) [reply gcp/dotscience-gcp.tf:105]: Sounds good. Majority of our changes to the ami is actually the startup script. Decoupling that and pushing that to S3, google storage means the images will only change when we need to update the version of local-stack/env. file changes. Maybe even that can be extracted out!

**Review by prisamuel — APPROVED** [review]

File: gcp/dotscience-gcp.tf (hunk 1 above)
Comment (rusenask) [reply gcp/dotscience-gcp.tf:105]: so maybe we can move startup.go to a separate repo?

Comment (prisamuel) [reply gcp/dotscience-gcp.tf:105]: Yeah, why not? We would need to update both the tf environments to use it.

**Pull request merged**